inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_binary_elementwise.cpp

```diff
@@ -48,9 +49,51 @@ void processBinaryEltwise(std::shared_ptr<ngraph::Node> eltwise, size_t lhsIndex
         broadcastInput = std::make_shared<ngraph::opset3::Concat>(ngraph::OutputVector{broadcastConst, broadcastInput}, 0);
     }
 
-    const auto shape = std::make_shared<ngraph::opset3::Maximum>(lhsInput, rhsInput);
+    const auto& lhsInputShape = lhsInput.get_partial_shape();
+    const auto& rhsInputShape = rhsInput.get_partial_shape();
 
-    auto outDSR = std::make_shared<ngraph::vpu::op::DynamicShapeResolver>(copied, shape);
+    VPU_THROW_UNLESS(lhsInputShape.is_static() && lhsInputShape.rank().get_length() == 1,
+        "DynamicToStaticShape transformation for {} of type {} expects lhs input shape to be static 1D vector, actual is {}",
+        eltwise->get_friendly_name(), eltwise->get_type_info(), lhsInputShape);
+
+    VPU_THROW_UNLESS(rhsInputShape.is_static() && rhsInputShape.rank().get_length() == 1,
+        "DynamicToStaticShape transformation for {} of type {} expects rhs input shape to be static 1D vector, actual is {}",
+        eltwise->get_friendly_name(), eltwise->get_type_info(), rhsInputShape);
+
+    std::shared_ptr<ngraph::Node> maxShape = std::make_shared<ngraph::opset3::Maximum>(lhsInput, rhsInput);
+    const auto& maxShapePartialShape = maxShape->output(0).get_partial_shape();
+    VPU_THROW_UNLESS(maxShapePartialShape.is_static(),
+        "DynamicToStaticShape transformation for {} of type {} expects output shape to have static rank, actual is dynamic");
+    const auto& maxShapeRank = ngraph::shape_size(maxShapePartialShape.to_shape());
+
+    const auto& updateOutputShapeOnZerosFrom = [&maxShape, &maxShapeRank, &shapeElementType](const ngraph::Output<ngraph::Node>& input) {
+        const auto& shapeValue = input.get_partial_shape();
+        const auto& rank = ngraph::shape_size(shapeValue.to_shape());
+
+        const auto& zeros = ngraph::opset5::Constant::create(shapeElementType, {rank}, std::vector<std::int64_t>(rank, 0));
```

Comment:
I think it's better to use the latest opset if possible which is `opset6`

Reply:
Fixed